From ltsi-dev-bounces@lists.linuxfoundation.org Mon Jun 18 04:06:00 2012
From: Tetsuyuki Kobayashi <koba@kmckk.co.jp>
Date: Mon, 18 Jun 2012 20:06:02 +0900
Subject: LTSI: ARM: shmobile: kzm9g: Support Real Time Clock
To: ltsi-dev@lists.linuxfoundation.org
Message-ID: <4FDF0B9A.6030404@kmckk.co.jp>


KZM-A9-GT board has RTC device r2025d at I2C channel 0
with slave address 0x64.

This patch enables reading and writing hardware real time clock on
KZM-A9-GT board. Interrupt is not yet supported.

How to test this patch using user space command.

Read RTC
 # hwclock -r

Copy system clock to RTC
 # hwclock -w

Shutdown and power off the board, and wait a while, then power on again.
Check RTC still keeps correct date.

Signed-off-by: Tetsuyuki Kobayashi <koba@kmckk.co.jp>
---
 arch/arm/configs/kzm9g_defconfig     |    1 +
 arch/arm/mach-shmobile/board-kzm9g.c |    3 +++
 2 files changed, 4 insertions(+)

--- a/arch/arm/configs/kzm9g_defconfig
+++ b/arch/arm/configs/kzm9g_defconfig
@@ -108,6 +108,7 @@ CONFIG_MMC_SH_MMCIF=y
 CONFIG_NEW_LEDS=y
 CONFIG_LEDS_CLASS=y
 CONFIG_RTC_CLASS=y
+CONFIG_RTC_DRV_RS5C372=y
 CONFIG_DMADEVICES=y
 CONFIG_SH_DMAE=y
 CONFIG_ASYNC_TX_DMA=y
--- a/arch/arm/mach-shmobile/board-kzm9g.c
+++ b/arch/arm/mach-shmobile/board-kzm9g.c
@@ -278,6 +278,9 @@ static struct pcf857x_platform_data pcf8
 static struct i2c_board_info i2c0_devices[] = {
 	{
 		I2C_BOARD_INFO("ak4648", 0x12),
+	},
+	{
+		I2C_BOARD_INFO("r2025sd", 0x32),
 	}
 };
 
